### PR TITLE
Hide chat panel button when not in a call

### PR DESCRIPTION
**Pull request by cole-miller**
cc @nathansobo 

Release Notes:

- Hide chat panel button by default when not in a call

### CODE REVIEW

File: crates/collab_ui/src/chat_panel.rs [hunk 1]
```diff
@@ -1096,6 +1096,10 @@ impl FocusableView for ChatPanel {
 }
 
 impl Panel for ChatPanel {
+    fn should_show_button(&self, cx: &gpui::WindowContext) -> bool {
```
Comment (maxdeviant) [inline crates/collab_ui/src/chat_panel.rs:1099]: Currently we return `None` in icon if we need to hide a panel button.

Not sure if we need to add a new method for it. Perhaps it conveys the intent better.

Either way, we should be consistent for the various panels.

Comment (cole-miller) [reply crates/collab_ui/src/chat_panel.rs:1099]: I'll switch this to the existing convention, thanks for flagging

**Comment by nathansobo** [comment]
Nice! Thanks!

**Comment by cole-miller** [comment]
This is now available as an additional value `"when_in_call"` for the `chat_panel.button` setting (which is the new default). I've also extended that field a bit to accept `"always"` and `"never"` which are equivalent to the boolean values it could take before (and can still take, for backward compatibility).

**Comment by cole-miller** [comment]
Noted while testing something unrelated: this needs a touch-up to bring the button back when the chat panel is manually opened e.g. by an action.

**Pull request merged**